langchain4j-coherence/src/main/java/dev/langchain4j/store/embedding/coherence/CoherenceMetadataFilterMapper.java

```diff
@@ -0,0 +1,134 @@
+package dev.langchain4j.store.embedding.coherence;
+
+import com.oracle.coherence.ai.DocumentChunk;
+import com.tangosol.util.Extractors;
+import com.tangosol.util.Filters;
+import com.tangosol.util.ValueExtractor;
+import dev.langchain4j.store.embedding.filter.Filter;
+import dev.langchain4j.store.embedding.filter.comparison.IsEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsGreaterThan;
+import dev.langchain4j.store.embedding.filter.comparison.IsGreaterThanOrEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsIn;
+import dev.langchain4j.store.embedding.filter.comparison.IsLessThan;
+import dev.langchain4j.store.embedding.filter.comparison.IsLessThanOrEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsNotEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsNotIn;
+import dev.langchain4j.store.embedding.filter.logical.And;
+import dev.langchain4j.store.embedding.filter.logical.Not;
+import dev.langchain4j.store.embedding.filter.logical.Or;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+/**
+ * A mapper that maps {@link Filter LangChain filters} to
+ * {@link com.tangosol.util.Filter Coherence filters} that
+ * can be applied to {@link DocumentChunk} metadata.
+ */
+class CoherenceMetadataFilterMapper
+    {
+    /**
+     * Return the Coherence filter that is equivalent to the
+     * specified LangChain filter.
+     *
+     * @param filter  the LangChain filter to convert
+     *
+     * @return the equivalent Coherence filter
+     */
+    static com.tangosol.util.Filter<DocumentChunk> map(Filter filter) {
+        if (filter == null) {
+            return null;
+        } else if (filter instanceof IsEqualTo) {
+            return mapEqual((IsEqualTo) filter);
```

Comment:
minor: as we already migrated to JDK17, we can use [pattern matching for instanceof](https://openjdk.org/jeps/394) now.